File: giokit-no-op/src/main/java/com/growingio/giokit/GioKit.java
package com.growingio.giokit;

import android.app.Activity;
import android.app.Application;

/**
 * <p>
 * 对外接口
 *
 * @author cpacm 2022/3/8
 */
public class GioKit {

    private GioKit() {
    }

    public static Builder with(Application application) {
        return new Builder(application);
    }

    public static class Builder {
        Builder(Application app) {
        }

        public Builder attach(boolean attach) {
            return this;
        }

        public Builder setGMonitorOption(Object option) {
            return this;
        }

        public void build() {
        }
    }

    public static void attach(Activity activity) {
    }

    public static void detach(Activity activity) {
    }
}

File: giokit-no-op/src/main/java/com/growingio/giokit/GioKit.kt
package com.growingio.giokit

import android.app.Application

/**
 * <p>
 *     对外 API 接口
 * @author cpacm 2021/8/11
 */
public class GioKit private constructor() {

    companion object {
        const val TAG = "GIOKIT"

        @JvmStatic
        fun with(app: Application): Builder {
            return Builder(app)
        }
    }

    class Builder(private val app: Application) {
        fun build() {
        }
    }
}